chia/data_layer/data_store.py

```diff
@@ -1411,11 +1456,15 @@ async def write_tree_to_file(
             merkle_blob = await self.get_merkle_blob(root.node_hash)
         if hash_to_index is None:
             hash_to_index = merkle_blob.get_hashes_indexes()
+        if existing_hashes is None:
+            previous_generation = max(0, root.generation - 1)
```

Comment:
so this `max()` is special casing the first generation which (presently when using our code anyways) is always empty?  maybe it would be worth being more explicit about handling that case and what we'll do?

```python
if root.generation == 0:
    existing_hashes = {}
```

or somesuch?